File: pset6/readability/readability.py
from cs50 import get_string
answer = lettercount = sentences = wordcount = grade = i = 0

prompt = get_string("Text: ")
value = len(prompt)
for i in range(value):
    if prompt[i].islower() or prompt[i].isupper() == True:
        lettercount += 1

# check if value is a space or not
for i in range(value):
    if (i == 0 and prompt[i] != " ") or (i != value and prompt[i] == " " and prompt[i + 1] != " ") == True:
        wordcount += 1
    elif prompt[i] == '.' or prompt[i] == '!' or prompt[i] == '?':
        sentences += 1

for i in range(value):
    L = float(lettercount * 100) / (wordcount)
    S = float(sentences * 100) / (wordcount)
    answer = 0.0588 * L - 0.296 * S - 15.8
    grade = round(answer)

if grade >= 16:
    print("Grade 16+")
elif grade < 1:
    print("Before Grade 1")
else:
    print(f"Grade {grade}")

File: pset2/readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

float index = 0, lettercount = 0, sentences = 0, wordcount = 1;
int grade = 0;

int main(void)
{
    string prompt = get_string("Text: ");

    for (int i = 0; i < strlen(prompt); i++)
    {
        if (prompt[i] >= 97 && prompt[i] <= 122)
        {
            lettercount++;
        }

        if (prompt[i] >= 65 && prompt[i] <= 90)
        {
            lettercount++;
        }
    }

    for (int i = 0; prompt[i] != '\0'; i++)
    {
        if (prompt[i] == ' ' && prompt[i + 1] != ' ')
        {
            wordcount++;
        }
    }
    for (int i = 0; prompt[i] != '\0'; i++)
    {
        if (prompt[i] == '.' || prompt[i] == '!' || prompt[i] == '?')
        {
            sentences++;
        }
    }

    // You might want to cast your one or more values to float before performing division when calculating L and S!

    {
        float L = (lettercount * 100) / (wordcount);
        float S = (sentences * 100) / (wordcount);
        index = 0.0588 * L - 0.296 * S - 15.8;
        grade = round(index);
    }

    if (grade >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (grade < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", grade);
    }
}


